rocketpy/Components.py

```diff
@@ -0,0 +1,125 @@
+__author__ = "Mateus Stano Junqueira"
+__copyright__ = "Copyright 20XX, RocketPy Team"
+__license__ = "MIT"
+
+from collections import namedtuple
+
+
+class Components:
+    """A class to hold components of the Rocket class. A component is
+    an object that added to the rocket given a position relative to the
+    rocket's coordinate system origin. Components can be added to the
+    rocket using the 'add' methods. This class is currently used only to
+    hold aerodynamic surfaces.
+
+    Attributes
+    ----------
+    _components: list
+        A list of named tuples containing the component and its position.
+    component_tuple: namedtuple
+        A named tuple containing the component and its position.
+    """
+
+    def __init__(self):
+        """Initialize an empty components list instance."""
+        self.component_tuple = namedtuple("component_tuple", "component position")
+        self._components = []
+
+    def __repr__(self):
+        """Return a string representation of the Components instance."""
+        return repr(self._components)
+
+    def __len__(self):
+        """Return the number of components in the collection."""
+        return len(self._components)
+
+    def __getitem__(self, index):
+        """Return the number of components in the collection."""
+        return self._components[index]
+
+    def __iter__(self):
+        """Return an iterator over the components."""
+        return iter(self._components)
+
+    def add(self, component, position):
+        """Add a component to the collection.
+
+        Parameters
+        ----------
+            component: AeroSurface, Motor
+                The component to be added to the rocket.
+            position: int, float
+                The position of the component relative to the rocket's
+                coordinate system origin.
+
+        Returns
+        -------
+        None
+        """
+        self._components.append(self.component_tuple(component, position))
+
+    def get_by_type(self, component_type):
+        """Get the components of a specified given type.
```

Comment:
I would recommed """Search the components list (or colletions?) and return a list with all the components of a given type